src/subplots.ts

```diff
@@ -1688,116 +1774,161 @@ export class BasePlot extends PlotData {
       var mouse3X = 0; var mouse3Y = 0;
       var canvas = document.getElementById(this.canvas_id);
-      var ctrlKey = false;
-      window.addEventListener('keydown', e => {if (e.key == "Control") {ctrlKey = true}});
+      var ctrlKey = false; var shiftKey = false;
+      var zoomBox = new SelectionBox();
+      window.addEventListener('keydown', e => {
+        if (e.key == "Control") ctrlKey = true;
+        if (e.key == "Shift") {
+          shiftKey = true;
+          if (!ctrlKey) { this.isSelecting = true; canvas.style.cursor = 'crosshair'; this.draw() };
+        }
+      });
 
-      window.addEventListener('keyup', e => {if (e.key == "Control") {ctrlKey = false}});
+      window.addEventListener('keyup', e => {
+        if (e.key == "Control") ctrlKey = false;
+        if (e.key == "Shift") { shiftKey = false; this.isSelecting = false; canvas.style.cursor = 'default'; this.draw() };
+      });
 
       canvas.addEventListener('mousemove', e => {
         [canvasMouse, frameMouse, absoluteMouse] = this.projectMouse(e);
         this.mouseMove(canvasMouse, frameMouse, absoluteMouse);
+        if (this.isZooming) canvas.style.cursor = 'crosshair';
         if (this.interaction_ON) {
           if (isDrawing) {
-            if (!clickedObject?.mouseMove(canvasDown, canvasMouse)) {
-              canvas.style.cursor = 'move';
-              this.translation = this.mouseTranslate(canvasMouse, canvasDown);
+            if (clickedObject instanceof SelectionBox != true) { // TODO: BEURK !!!!
+              if (!clickedObject?.mouseMove(canvasDown, canvasMouse) && !this.isSelecting && !this.isZooming) {
+                canvas.style.cursor = 'move';
+                this.translation = this.mouseTranslate(canvasMouse, canvasDown);
+              }
             }
           }
           this.draw();
         }
+        if (this.isSelecting) {
+          if (isDrawing) {
+            if (clickedObject instanceof SelectionBox) {
+              clickedObject.mouseMove(frameDown, frameMouse);
+              this.updateSelectionBox(clickedObject.minVertex, clickedObject.maxVertex);
+            }
+            else this.updateSelectionBox(frameDown, frameMouse);
+          }
+        }
+        if (this.isZooming) {
+          if (isDrawing) this.drawZoomBox(zoomBox, frameDown, frameMouse, this.context_show);
+        }
         const mouseInCanvas = (e.offsetX >= this.X) && (e.offsetX <= this.width + this.X) && (e.offsetY >= this.Y) && (e.offsetY <= this.height + this.Y);
         if (!mouseInCanvas) { isDrawing = false };
       });
 
       canvas.addEventListener('mousedown', e => {
         [canvasDown, frameDown, clickedObject] = this.mouseDown(canvasMouse, frameMouse, absoluteMouse);
-        if (clickedObject instanceof newAxis) this.is_drawing_rubber_band = true
-        else this.is_drawing_rubber_band = false
+        if (clickedObject instanceof newAxis || this.isSelecting) this.is_drawing_rubber_band = true
```

Comment:
Done the change

To answer to you, maybe yes or no. The mouse_interaction method must be totally re-think and maybe built from scratch, because it is not good now